### DISCUSSION TITLE
Dynamic routing

### DISCUSSION THREAD

**Discussion by jmfk** [google/mesop#1116] [Ideas]
I might be missing something but I cant find a way to dynamically route things (say for a blog I use a slug as part of the URL)?

**Comment by jmfk** [comment]
Correct me if I'm wrong. It seems like this would be possible by replacing the _path_to_page_config dict (in runtime) with something that could search for regex (e.g. something like "/blogs/(.[^/])/" ) and annotate the config object with the matching values from the regex?
Are there any more things that would need to be added to make this work?
Are there any other plans for path routing?

**Comment by richard-to** [comment]
Looks like no. We did have an issue for it here: https://github.com/google/mesop/issues/514, but have closed it as not planned.

But I think if we get enough requests for this feature, we can always reopen it and add it to Mesop. I do think it is a useful feature, but not too high priority for Mesop's current use cases where people typically building internal demos and tools. I think if people were building more public web applications, then having the nicer urls would be more beneficial.

I think the only current option is to user query params, such as `blogs?id=X`. Definitely not as nice though.

**Reply by jmfk** [reply]
Ok maybe I submit a pull request. I don't need this right now (because I'm staying with Remix.app for my web stuff, due to the need for a server rendered solution, given a solution to pre-render HTML on the server (which I don't think is straight forward?) Mesop would be an viable alternative to server rendered React (such as Remix or Next.js). I actually find Mesop much faster and less cluttered to develop in (I also find Python easier to work with then Typescript and "node").
